### DISCUSSION TITLE
Gradle Build Failed

### DISCUSSION THREAD

**Discussion by Chandra-Sekhar-Bala** [oppia/oppia-android#4922] [Q&A (Installation)]
### Operating System

WIndows

### Android Studio Version

Electric Eel

### RAM size

8

### Free disk space

80GB

### Which step are you stuck on?

Followed all the process written [here](https://github.com/oppia/oppia-android/wiki/Installing-Oppia-Android). Still this error pops up.

### Error Log

```bash
> Task :utility:kaptGenerateStubsDebugKotlin FAILED
Execution failed for task ':utility:kaptGenerateStubsDebugKotlin'.
> Could not resolve all files for configuration ':utility:debugCompileClasspath'.
   > Failed to transform model.jar (project :model) to match attributes {artifactType=android-classes, org.gradle.category=library, org.gradle.dependency.bundling=external, org.gradle.jvm.version=15, org.gradle.libraryelements=jar, org.gradle.usage=java-api}.
      > Execution failed for JetifyTransform: E:\Android\open-source\oppia-android\model\build\libs\model.jar.
         > Failed to transform 'E:\Android\open-source\oppia-android\model\build\libs\model.jar' using Jetifier. Reason: Unsupported class file major version 59. (Run with --stacktrace for more details.)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

### Approaches already used to resolve the issue

Deleted the project and re-imported from github.

**Comment by MohitGupta121** [comment]
Hey @Chandra-Sekhar-Bala, are you updating the Kotlin or SDK version while project set up?
And can you please try in [Android Studio Bumblebee - Patch 3](https://developer.android.com/studio/archive) (Oppia works well in that version)?

Thank You.

**Reply by InquisitiveCatherine** [reply]
Hi @MohitGupta121 I am facing the same error here. I tried installing Android Studio Bumblebee - Patch 3 and also re-cloned the repository yet the same issue persists.

**Reply by MohitGupta121** [reply]
@InquisitiveCatherine you are also using windows?

**Reply by MohitGupta121** [reply]
Hey @Chandra-Sekhar-Bala, I requested you to recheck that you not changing Kotlin version of oppia-android while setup or building the project. That's why this error happens.

**Reply by InquisitiveCatherine** [reply]
> @InquisitiveCatherine you are also using windows?
@MohitGupta121  Yes, I am using Windows 11

**Reply by MohitGupta121** [reply]
Okay Thanks for confirming @InquisitiveCatherine , can you please cross-check kotlin version from oppia-android codebase in github and in your local. Thanks!

**Reply by InquisitiveCatherine** [reply]
> Okay Thanks for confirming @Chandra-Sekhar-Bala, can you please cross-check kotlin version from oppia-android codebase in github and in your local. Thanks!

I am using ext.kotlin_version = '1.4.20' in my build.gradle file. I feel it's same as the one in oppia-android. 
If you are referring to that.
![image](https://github.com/oppia/oppia-android/assets/133565952/35032bb0-0c4c-46a6-b76b-c86b98567029)

**Reply by MohitGupta121** [reply]
Okay thanks @InquisitiveCatherine  for confirming. It's wondering for me why then this error happens, let me cross check with other windows user. Thanks

**Reply by MohitGupta121** [reply]
This issue is related to our use of jetifier currently. I believe it will be resolved with the upcoming build platform changes, but to explain a bit:

In gradle.properties, `android.enableJetifier=true` automatically converts third-party libraries to use AndroidX. 
The maximum supported major version for jetifier class file format in Android is 58, which corresponds to Java 14.
The `model.jar` was compiled with Java 15/major version 59, hence the incompatibility.

A fix would be to compile the JAR file with a lower version of Java. 

@InquisitiveCatherine can you please check using https://developer.android.com/studio/intro/studio-config#jdk.

Solution by - @adhiamboperes 

Edited to add: looks like you're using JDK 15--this isn't supported with our version of Gradle. You'll need to install JDK 8 LTS for it to work correctly (9 is fine too).

**Reply by InquisitiveCatherine** [reply]
> This issue is related to our use of jetifier currently. I believe it will be resolved with the upcoming build platform changes, but to explain a bit:
> 
> In gradle.properties, `android.enableJetifier=true` automatically converts third-party libraries to use AndroidX. The maximum supported major version for jetifier class file format in Android is 58, which corresponds to Java 14. The `model.jar` was compiled with Java 15/major version 59, hence the incompatibility.
> 
> A fix would be to compile the JAR file with a lower version of Java.
> 
> @InquisitiveCatherine can you please check using https://developer.android.com/studio/intro/studio-config#jdk.
> 
> Solution by - @adhiamboperes

Hi @MohitGupta121 . Thanks a lot. Lowering the JDK version got the build to work.
Once again Thanks 🙌🙌

**Reply by jkugsiya** [reply]
Hi @InquisitiveCatherine, I am also facing the same issue and tried lowering the jdk version to jdk 8 but still getting the issue. You can refer to the screenshot below:

![image](https://github.com/oppia/oppia-android/assets/42642325/b3f6f099-f3ea-41fe-9ce5-2c034ecc59b5)

**Reply by adhiamboperes** [reply]
Hi @jkugsiya, your current stack trace appears different from the original one.
Once you can confirm that the Java version is correct set, please clean and rebuild your app.

**Comment by MohitGupta121** [comment]
@Chandra-Sekhar-Bala If your query is solved please mark it as an answer to the above comment. So that it is helpful to others in the community. Otherwise, we have to mark this discussion as `stale`. If you have any queries/doubts please feel free to ask in this thread.
Thanks.
